Rest/Controller/ContentController.php

```diff
@@ -0,0 +1,229 @@
+                    } elseif (key($fieldValue) == 'destinationContentId') {
+                        $fieldValue = $this->getRelatedFields((string) $fieldValue, $language);
+                        $field = 'image';
+                    } elseif ($field == 'image') {
+                        $fieldObj = $contentValue->getFieldsByLanguage($language);
+                        $fieldValue = $this->imageVariations($fieldObj[$field], $contentValue->versionInfo, $this->request->get('image'));
+                    }
+
+                    $content[$contentId]['fields'][] = array(
+                        'key' => $field,
+                        'value' => (string) $fieldValue,
+                    );
+                }
+            }
+        }
+
+        return $content;
+    }
+
+    /**
+     * Checks if fields are given, if not - returns all of them.
+     *
+     * @param string                                                    $fields
+     * @param \eZ\Publish\API\Repository\Values\ContentType\ContentType $contentType
+     *
+     * @return array|null
+     */
+    protected function prepareFields(ContentType $contentType, $fields = null)
+    {
+        if (null !== $fields) {
+            return explode(',', $fields);
+        }
+
+        $fields = array();
+        $contentFields = $contentType->getFieldDefinitions();
+
+        foreach ($contentFields as $field) {
+            $fields[] = $field->identifier;
+        }
+
+        return $fields;
+    }
+
+    /**
+     * Returns image uri based on variation provided in url.
+     * If none is set original is returned.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Field        $fieldValue
+     * @param \eZ\Publish\Core\Repository\Values\Content\VersionInfo $versionInfo
+     * @param string|null                                            $variation
+     *
+     * @return string
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidVariationException
+     * @throws \eZ\Publish\Core\MVC\Exception\SourceImageNotFoundException
+     */
+    private function imageVariations(Field $fieldValue, $versionInfo, $variation = null)
+    {
+        if (!isset($fieldValue->value->id)) {
+            return '';
+        }
+
+        $variations = $this->configResolver->getParameter('image_variations');
+
+        if ((null === $variation) || !in_array($variation, array_keys($variations))) {
+            $variation = 'original';
+        }
+
+        return $this->imageVariationService->getVariation($fieldValue, $versionInfo, $variation)->uri;
+    }
+
+    /**
+     * Return uri of the related image field.
+     *
+     * @param mixed  $contentId
+     * @param string $language
+     *
+     * @return string
+     */
+    private function getRelatedFields($contentId, $language)
+    {
+        $content = $this->contentService->loadContent($contentId);
+        $fieldObj = $content->getFieldsByLanguage($language);
+
+        if (!isset($fieldObj['image'])) {
```

Comment:
Same here, it won't work for ezimage fieldtype with identifier different then "image"